src/stream/src/common/log_store_impl/kv_log_store/reader.rs

```diff
@@ -74,13 +84,28 @@ impl<S: StateStore> KvLogStoreReader<S> {
             state_store,
             serde,
             rx,
+            read_flushed_chunk_future: None,
             first_write_epoch: None,
             state_store_stream: None,
             latest_offset: TruncateOffset::Barrier { epoch: 0 },
             truncate_offset: TruncateOffset::Barrier { epoch: 0 },
             metrics,
         }
     }
+
+    async fn may_continue_read_flushed_chunk(
+        &mut self,
+    ) -> LogStoreResult<Option<(ChunkId, StreamChunk, u64)>> {
+        if let Some(future) = self.read_flushed_chunk_future.as_mut() {
+            let result = future.await;
+            self.read_flushed_chunk_future
+                .take()
+                .expect("future not None");
```

Comment:
This `expect` panics when the option "is" None. 👀

And it will be a lint error in the future:
```rust
error: unused pinned boxed `futures::Future` trait object that must be used
   --> src/stream/src/common/log_store_impl/kv_log_store/reader.rs:102:13
    |
102 | /             self.read_flushed_chunk_future
103 | |                 .take()
104 | |                 .expect("future not None");
    | |__________________________________________^
    |
    = note: futures do nothing unless you `.await` or poll them
    = note: requested on the command line with `-F unused-must-use`
```

Reply:
Should we change it to 

```rust
_ = self.read_flushed_chunk_future
  .take()
  .expect("future is None");
```

or can we simply
```rust
self.read_flushed_chunk_future = None
```
?

I'm not sure what you are going to do here.